internal/provider/resource_vault_cluster.go

```diff
@@ -448,3 +514,60 @@ func resourceVaultClusterImport(ctx context.Context, d *schema.ResourceData, met
 
 	return []*schema.ResourceData{d}, nil
 }
+
+func inPlusTier(tier string) bool {
+	return tier == string(vaultmodels.HashicorpCloudVault20201125TierPLUSSMALL) ||
+		tier == string(vaultmodels.HashicorpCloudVault20201125TierPLUSMEDIUM) ||
+		tier == string(vaultmodels.HashicorpCloudVault20201125TierPLUSLARGE)
+}
+
+func validatePerformanceReplicationChecksAndReturnPrimaryIfAny(ctx context.Context, client *clients.Client, d *schema.ResourceData) ([]diag.Diagnostic, *vaultmodels.HashicorpCloudVault20201125Cluster) {
+	primaryClusterLinkStr := getPrimaryLinkIfAny(d)
+	// if no primary_link has been supplied, then treat this as as single cluster creation.
+	if primaryClusterLinkStr == "" {
+		return nil, nil
+	}
+
+	primaryClusterLink, err := buildLinkFromURL(primaryClusterLinkStr, VaultClusterResourceType, client.Config.OrganizationID)
+	if err != nil {
+		return diag.Errorf("invalid primary_link supplied %v", err), nil
+	}
+
+	primaryCluster, err := clients.GetVaultClusterByID(ctx, client, primaryClusterLink.Location, primaryClusterLink.ID)
+	if err != nil {
+		if clients.IsResponseCodeNotFound(err) {
+			return diag.Errorf("primary cluster (%s) must exist", primaryClusterLink.ID), nil
+
+		}
+		return diag.Errorf("unable to check for presence of an existing primary Vault cluster (%s): %v", primaryClusterLink.ID, err), nil
+	}
+
+	if !inPlusTier(string(primaryCluster.Config.Tier)) {
+		return diag.Errorf("primary cluster (%s) must be plus-tier", primaryClusterLink.ID), primaryCluster
+	}
+
+	// tier should be specified, even if secondary inherits it from the primary cluster.
+	if !strings.EqualFold(d.Get("tier").(string), string(primaryCluster.Config.Tier)) {
+		return diag.Errorf("a secondary's tier must match that of its primary (%s)", primaryClusterLink.ID), primaryCluster
+	}
+
+	if primaryCluster.PerformanceReplicationInfo != nil && primaryCluster.PerformanceReplicationInfo.Mode == vaultmodels.HashicorpCloudVault20201125ClusterPerformanceReplicationInfoModeSECONDARY {
+		return diag.Errorf("primary cluster (%s) is already a secondary", primaryClusterLink.ID), primaryCluster
+	}
+
+	// min_vault_version is not actually set in the terraform state, however it can force a new deployment. hence, defend against cases where it can
```

Comment:
Unclear and should be simplified. What's the user's key takeaway from this comment?

Reply:
Updated